Classes/AccessibleObject.php

```diff
@@ -83,6 +84,9 @@ public function offsetGet($propertyName)
             return (boolean)call_user_func([$this->object, $propertyName]);
         }
         $result = ObjectAccess::getProperty($this->object, $propertyName);
+        if ($result instanceof Collection) {
+            return new IterableAccessibleObject($result->toArray());
+        }
```

Comment:
Thanks for this, but that would effectively add a dependency to doctrine I guess.
I would suggest to replace this with
```php
if ($result instanceof \IteratorAggregate) {
    return new IterableAccessibleObject($result->getIterator());
}
```

Reply:
Makes sense, i changed and tested it